dts/arm/st/f0/stm32f031.dtsi

```diff
@@ -28,5 +28,10 @@
 				#pwm-cells = <3>;
 			};
 		};
+		bbram: bbram@40002850 {
+			compatible = "st,stm32-bbram";
+			reg = <0x40002850 0x14>;
+			status = "disabled";
+		};
```

Comment:
Rather than adding dedicated nodes and adding a dependency on `COUNTER_RTC_STM32`, I'd prefer this is added as a child of existing RTC nodes, in the same way we're proceeding with PWM and timers and in lin with with is proposed in this [new RTC API proposal](https://github.com/zephyrproject-rtos/zephyr/pull/52618)

Reply:
Done. Please check if it is correct now.